lib/echo/echo_client.rb

```diff
@@ -34,10 +34,39 @@ def get_phones(user_id, token)
       get("/echo-rest/users/#{user_id}/phones.json", {}, token_header(token))
     end
 
-    def get_preferences(user_id, token)
+    def general_contact_valid?(contact)
+      return false if contact.blank? || contact['phones'].blank? || contact['address'].blank?
+      true
+    end
+
+    def get_preferences(user_id, token, client, access_token)
       response = get("/echo-rest/users/#{user_id}/preferences.json", {}, token_header(token))
       if response.status == 404
         response = update_preferences(user_id, {preferences: {}}, token)
+      elsif response.status == 200 && !general_contact_valid?(response.body['preferences']['general_contact'])
+        user_response = get_echo_user(user_id, token)
```

Comment:
One call to get ECHO user, if it is a first-time EDSC user, get ECHO empty user obj with GUID. The other one get the content of the user from URS. Then merge the URS info to the ECHO user.